scripts/upload_to_sds.py

```diff
@@ -31,28 +68,61 @@ def main():
         default=pathlib.Path(".env"),
         help="Path to .env file containing SDS_SECRET_TOKEN",
     )
+    parser.add_argument(
+        "--channel",
+        help="Channel to use for creating capture after upload",
+    )
+    parser.add_argument(
+        "--create-capture",
+        action="store_true",
+        help="Create a capture after uploading files",
+    )
     args = parser.parse_args()
 
     data_dir = args.data_dir.resolve()
-    if not data_dir.exists():
-        msg = f"Data directory '{data_dir}' does not exist!"
-        raise ValueError(msg)
+    if not data_dir.exists() or not data_dir.is_dir():
+        logger.error(f"Data directory '{data_dir}' does not exist or is not a directory!")
+        sys.exit(1)
+    if not os.access(data_dir, os.R_OK):
+        logger.error(f"Data directory '{data_dir}' is not readable!")
+        sys.exit(1)
 
     sds = Client(
         host="sds.crc.nd.edu",
         env_file=args.dotenv,
     )
     sds.dry_run = args.dry_run
-
     sds.authenticate()
-    sds.upload(
+
+    upload_results = sds.upload(
         local_path=data_dir,
         sds_path=args.reference_name,
         verbose=True,
     )
 
+    if (upload_results):
+        success_results = [success for success in upload_results if success]
+        failed_results = [success for success in upload_results if not success]
+        print(f"Uploaded {len(success_results)} assets.")
+        print(f"{len(failed_results)} assets failed.")
+    else:
+        print("Upload results are not available.")
+
     if sds.dry_run:
         print("Turn off dry-run to actually upload files!")
+    else:
+        if args.create_capture:
+            if not args.channel:
+                logger.error("--channel is required when --create-capture is used")
+                sys.exit(1)
+            logger.info("Upload completed successfully. Creating capture...")
+            success = create_capture(sds, args.reference_name, args.channel)
+            if success:
+                logger.info("Capture created successfully!")
+            else:
+                logger.error("Failed to create capture")
+                sys.exit(1)
+
```

Comment:
[nit] you can remove an indentation level by adding `return None` after the print, then outdent the block below (i.e. removing `else`)